File: solutions/java/src/trafficsignalsystem/Road.java
package trafficsignalsystem;

public class Road {
    private final String id;
    private final String name;
    private TrafficLight trafficLight;

    public Road(String id, String name) {
        this.id = id;
        this.name = name;
    }

    public void setTrafficLight(TrafficLight trafficLight) {
        this.trafficLight = trafficLight;
    }

    public TrafficLight getTrafficLight() {
        return trafficLight;
    }

    public String getId() {
        return id;
    }
}

File: solutions/c#/TrafficSignalSystem/Road.cs
namespace TrafficSignalSystem
{
    public class Road
    {
        public string Id { get; }
        public string Name { get; }
        public TrafficLight TrafficLight { get; private set; }

        public Road(string id, string name)
        {
            Id = id;
            Name = name;
        }

        public void SetTrafficLight(TrafficLight trafficLight)
        {
            TrafficLight = trafficLight;
        }
    }
}